packages/jest-snapshot/src/matcher.js

```diff
@@ -19,76 +20,89 @@ type CompareResult = {
   message: ?string,
 };
 
-module.exports = (
-  filePath: Path,
-  options: Object,
-  snapshotState: SnapshotState,
-) => (util: any, customEquality: any) => {
-  return {
-    negativeCompare() {
-      throw new Error(
-        'Jest: `.not` can not be used with `.toMatchSnapshot()`.',
-      );
-    },
-    compare(actual: any, expected: any): CompareResult {
-      if (expected !== undefined) {
+module.exports = {
+  jasmineMatcher: (
+    config: Config,
+    snapshotState: SnapshotState,
+  ) => (util: any, customEquality: any) => {
+    return {
+      negativeCompare() {
         throw new Error(
-          'Jest: toMatchSnapshot() does not accept parameters.',
+          'Jest: `.not` can not be used with `.toMatchSnapshot()`.',
         );
-      }
+      },
+      compare(actual: any, expected: any): CompareResult {
+        if (expected !== undefined) {
+          throw new Error(
+            'Jest: toMatchSnapshot() does not accept parameters.',
+          );
+        }
+
+        return matcher(actual, expected, {config, snapshotState});
+      },
+    };
+  },
+
+  jestMatcher: (
```

Comment:
let's call it matcher. This way you can simply put `matcher` here and it will work because of the object-short-notation feature in JS.